kion/resource_ou_cloud_access_role.go

```diff
@@ -178,29 +197,20 @@ func resourceOUCloudAccessRoleRead(ctx context.Context, d *schema.ResourceData,
 	}
 	item := resp.Data
 
-	data := make(map[string]interface{})
-	data["aws_iam_path"] = item.OUCloudAccessRole.AwsIamPath
-	if hc.InflateSingleObjectWithID(item.AwsIamPermissionsBoundary) != nil {
-		data["aws_iam_permissions_boundary"] = hc.InflateSingleObjectWithID(item.AwsIamPermissionsBoundary)
-	}
-	if hc.InflateObjectWithID(item.AwsIamPolicies) != nil {
-		data["aws_iam_policies"] = hc.InflateObjectWithID(item.AwsIamPolicies)
-	}
-	data["aws_iam_role_name"] = item.OUCloudAccessRole.AwsIamRoleName
-	if hc.InflateObjectWithID(item.AzureRoleDefinitions) != nil {
-		data["azure_role_definitions"] = hc.InflateObjectWithID(item.AzureRoleDefinitions)
+	data := map[string]interface{}{
+		"aws_iam_path":                 item.OUCloudAccessRole.AwsIamPath,
+		"aws_iam_role_name":            item.OUCloudAccessRole.AwsIamRoleName,
+		"aws_iam_permissions_boundary": hc.InflateSingleObjectWithID(item.AwsIamPermissionsBoundary),
+		"long_term_access_keys":        item.OUCloudAccessRole.LongTermAccessKeys,
+		"name":                         item.OUCloudAccessRole.Name,
+		"ou_id":                        item.OUCloudAccessRole.OUID,
+		"short_term_access_keys":       item.OUCloudAccessRole.ShortTermAccessKeys,
+		"web_access":                   item.OUCloudAccessRole.WebAccess,
+		"aws_iam_policies":             hc.InflateObjectWithID(item.AwsIamPolicies),
+		"azure_role_definitions":       hc.InflateObjectWithID(item.AzureRoleDefinitions),
+		"gcp_iam_roles":                hc.InflateObjectWithID(item.GCPIamRoles),
+		"users":                        hc.InflateObjectWithID(item.Users),
 	}
-	data["long_term_access_keys"] = item.OUCloudAccessRole.LongTermAccessKeys
-	data["name"] = item.OUCloudAccessRole.Name
-	data["ou_id"] = item.OUCloudAccessRole.OUID
-	data["short_term_access_keys"] = item.OUCloudAccessRole.ShortTermAccessKeys
-	if hc.InflateObjectWithID(item.UserGroups) != nil {
-		data["user_groups"] = hc.InflateObjectWithID(item.UserGroups)
-	}
-	if hc.InflateObjectWithID(item.Users) != nil {
-		data["users"] = hc.InflateObjectWithID(item.Users)
-	}
-	data["web_access"] = item.OUCloudAccessRole.WebAccess
```

Comment:
This does change the behavior so that `d.Set` could be called on `aws_iam_permissions_boundary`, `aws_iam_policies`, and other "InflateObjectWithID" values, whereas in the old code if `InflateObjectWithID` returned nil, we'd never call Set.

I don't know if this would have any consequence.  Can you confirm we are safe to make this change?

Was this change just made to simplify code, or was there a bug in the old code?

Reply:
It was to simplify and understand what was happening here as I made the same changes to add `gcp_iam_role` to both files.

`InflateObjectWithID` never returns `nil` but rather an empty slice.  
`InflateSingleObjectWithID` could return `nil`, but the `Set` method within the SDK handles `nil` values correctly, so based on what I understand, calling `d.Set` even when `InflateSingleObjectWithID` returns `nil` is safe and will not introduce errors.

That specific code is [here](https://github.com/hashicorp/terraform-plugin-sdk/blob/b655613c564ccd5aeaed17eab0266c37b09a9363/helper/schema/resource_data.go#L216-L219)